### PR TITLE
ISSUE-1370 Quorum queue for OpenPaaS extension

### LINKED ISSUE #1370: [MU] Quorum queue for Openpaas extension
If quorum queues are enabled

### PR DESCRIPTION
**Pull request by quantranhong1999**
need https://github.com/apache/james-project/pull/2548

resolve #1370

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
com.linagora.tmail.OpenPaasAmqpForwardAttributeIntegrationTest$WithOpenPaasModuleAndDirectRabbitMqExchange.mailetShouldWorkNormalWhenExchangeAlreadyExistsWithDifferentConfiguration(File)

image

**Comment by chibenwa** [comment]
`[ERROR] /home/jenkins/build/workspace/Tmail_build_PR-1375/tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/AmqpUri.java:136:69:  error: cannot find symbol`

File: james-project
Comment (quantranhong1999) [inline james-project:1]: latest james + keep @chibenwa hashing commit

File: tmail-backend/tmail-third-party/openpaas/src/test/java/com/linagora/tmail/OpenPaasAmqpForwardAttributeIntegrationTest.java [hunk 1]
```diff
@@ -134,26 +134,25 @@ void setUpJamesServer(@TempDir File temporaryFolder, String exchange) throws Exc
...
+                    new OpenPaasModule(), new AbstractModule() {
+                        @Provides
+                        @Singleton
+                        public RabbitMQConfiguration provideCommonRabbitMQConfiguration() throws URISyntaxException {
+                            return rabbitMQExtension.getRabbitMQ().getConfiguration();
+                        }
+
+                        @Provides
+                        @Singleton
+                        public OpenPaasConfiguration provideOpenPaasConfiguration()
```
Comment (quantranhong1999) [inline tmail-backend/tmail-third-party/openpaas/src/test/java/com/linagora/tmail/OpenPaasAmqpForwardAttributeIntegrationTest.java:146]: recipientsShouldBePublishedToAmqpWhenSendingEmailWithExchangeTypeFanoutConfiguration(File) – com.linagora.tmail.OpenPaasAmqpForwardAttributeIntegrationTest$WithOpenPaasModuleAndFanoutRabbitMqExchange
<1s
Error
Unable to create injector, see the following errors:
1) [Guice/BindingAlreadySet]: OpenPaasConfiguration was bound multiple times.

**Comment by Arsnael** [comment]
Actually does it make sense at all to have that extension for memory? As it relies on rabbitmq and memory does not use rabbitmq?

**Comment by quantranhong1999** [comment]
> Actually does it make sense at all to have that extension for memory? As it relies on rabbitmq and memory does not use rabbitmq?

I feel the same concern. I do not think we would use memory app + OpenPaaS in any real case. Maybe it is just to ease testing...
(should have used proper distributed app + OpenPaaS extension for integration test IMO)

> https://james-jenkins.lin-saas.com/job/Tmail%20build/job/PR-1375/16/testReport/junit/com.linagora.tmail.contact/OpenPaasContactsConsumerTest/openPaasContactsQueueShouldBeQuorumQueueWhenQuorumQueuesAreEnabled/
Related?

Likely test isolation issue. Let's me rebase.

**Comment by quantranhong1999** [comment]
> Related?

It was because of a breaking change in the OpenPaaS extension queue name:

https://github.com/linagora/tmail-backend/pull/1367/files#diff-149c084328a5edcae8366f34363133d17c3ad88b57ad3d79a23e6b2cd9371429R46

File: tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/configuration/OpenPaasConfiguration.java [hunk 2]
```diff
@@ -96,4 +101,9 @@ private static boolean readTrustAllSslCerts(Configuration configuration) {
             .orElse(OPENPAAS_REST_CLIENT_TRUST_ALL_SSL_CERTS_DISABLED);
     }
 
+    private static boolean readQuorumQueuesBypass(Configuration configuration) {
+        return Optional.ofNullable(configuration.getBoolean(OPENPAAS_QUEUES_QUORUM_BYPASS_PROPERTY, null))
+            .orElse(OPENPAAS_QUEUES_QUORUM_BYPASS_DISABLED);
```
Comment (HoussemNasri) [inline tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/configuration/OpenPaasConfiguration.java:106]:
```suggestion
return configuration.getBoolean(OPENPAAS_QUEUES_QUORUM_BYPASS_PROPERTY, OPENPAAS_QUEUES_QUORUM_BYPASS_DISABLED);
```

**Comment by quantranhong1999** [comment]
Green. One last review maybe?

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**